### ISSUE TITLE
Advanced v1.1 HW testing

### ISSUE THREAD

**Issue by elsalahy** [TheThingsIndustries/generic-node-se#36]
**Summary:** 

With the testing of the initial prototypes taking place in #28 

We need to move forward and do more advanced testing.
This issue is an umbrella for the planned activities and also separate issues can be opened and linked here.
We can close this issue after the activities are done and we attach a conclusion report.

**Why do we need this?**

To ensure a stable product before moving forward with the production prototypes.

**What is already there? What do you see now?**

Acceptance testing of the prototypes.

**What is missing? What do you want to see?**

- Plan advanced testing activities.
- FW libraries that support the advanced tests.
- Conclusion Report.

**How do you propose to implement this?**

- Plan
- Test
- Write-up conclusion

**Environment:**

Bare-metal

**Acceptance Criteria:**

Cover all essential product requirements.

**What can you do yourself and what do you need help with?**

Workload will be split between me and @azerimaker.

**Comment by azerimaker** [comment]
**Problem:** Currently testing the rest of the **v1.1** PCBs. There seem to be quite high number of boards (so far 8/60) with poor solder quality on Buck-Boost converter IC. Since it's a BGA package, pin connections are not visible. 

**Solution:** Reflowing (heating with hot air gun) and resoldering them seems to solve the problem. I'll also investigate the gerber and stencil files for potential improvements. 

FYI: @wienke @elsalahy

**Comment by wienke** [comment]
So this is due to poor manufacturing by Seeed?

**Comment by azerimaker** [comment]
> So this is due to poor manufacturing by Seeed?

I don't want to blatantly blame them, without doing any X-ray on every faulty boards, or properly analyzing the solder-paste stencils, but if I see that reflowing solves the problem in every faulty board, it's highly probable.

**Comment by azerimaker** [comment]
I've thoroughly investigated two faulty boards (№ 97, 98 on the list), which were not starting up at all. I could hear the buzzer clicking, which indicated that the buck-boost converter is trying to start-up then goes into protection mode. I've first desoldered the IC on board97 and observed that one solder ball was missing. After reballing the pad and resoldering it back the board worked flawlessly. 

![#97-1](https://user-images.githubusercontent.com/2662257/111038352-b975de80-8428-11eb-9475-5d3f584304b8.jpg)
![#97-2](https://user-images.githubusercontent.com/2662257/111038353-bbd83880-8428-11eb-8e6e-7c3a57ccd655.jpg)

Board98 was also showing the same behavior, so I followed the same procedure and as suspected this board also had missing or smaller than usual solder balls. 

![#098- (2)](https://user-images.githubusercontent.com/2662257/111038432-2a1cfb00-8429-11eb-92b0-47ea9c2ad923.jpg)

This IC (RP605) is extremely small (ball-pen for reference) and integrates lots of function in it, but it seems to be the main reason for the lower yield figure in our first beta run (#95). 

![#098- (5)](https://user-images.githubusercontent.com/2662257/111038433-2f7a4580-8429-11eb-9809-67096326a057.jpg)

**Possible solution:** 
To address this issue in the future we might consider using the bigger QFN package of this IC, but unfortunately it's only available with 3.3V and 2.0V outputs, but we're using 2.8V as our operating voltage.
I'll contact RICOH with regards to this. Maybe we can get a special manufacturing run of 2.8V version. Otherwise we can also consider increasing the operating voltage to 3.3V.
